src/safe_apps/tests/test_views.py

```diff
@@ -178,6 +219,126 @@ def test_apps_returned_on_same_key_pair(self) -> None:
+        json_response = [
+            {
+                "id": safe_app.app_id,
+                "url": safe_app.url,
+                "name": safe_app.name,
+                "iconUrl": safe_app.icon_url,
+                "description": safe_app.description,
+                "chainIds": safe_app.chain_ids,
+                "provider": None,
+                "accessControl": {
+                    "type": SafeApp.AccessControlPolicy.NO_RESTRICTIONS,
+                    "value": [],
+                },
+            }
+        ]
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.json(), json_response)
+
+    def test_apps_returned_on_empty_client_url(self) -> None:
+        safe_app = SafeAppFactory.create()
+        url = reverse("v1:safe-apps:list") + f'{"?client_url="}'
+
+        response = self.client.get(path=url, data=None, format="json")
+
+        json_response = [
+            {
+                "id": safe_app.app_id,
+                "url": safe_app.url,
+                "name": safe_app.name,
+                "iconUrl": safe_app.icon_url,
+                "description": safe_app.description,
+                "chainIds": safe_app.chain_ids,
+                "provider": None,
+                "accessControl": {
+                    "type": SafeApp.AccessControlPolicy.NO_RESTRICTIONS,
+                    "value": [],
+                },
+            }
+        ]
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.json(), json_response)
+
+    def test_apps_returned_on_same_client_url_key_pair(self) -> None:
+        client_1 = ClientFactory.create(url="safe.com")
+        client_2 = ClientFactory.create(url="pump.com")
+        safe_app_1 = SafeAppFactory.create(exclusive_clients=(client_1,))
+        SafeAppFactory.create(exclusive_clients=(client_2,))
+        json_response = [
+            {
+                "id": safe_app_1.app_id,
+                "url": safe_app_1.url,
+                "name": safe_app_1.name,
+                "iconUrl": safe_app_1.icon_url,
+                "description": safe_app_1.description,
+                "chainIds": safe_app_1.chain_ids,
+                "provider": None,
+                "accessControl": {
+                    "type": SafeApp.AccessControlPolicy.DOMAIN_ALLOWLIST,
+                    "value": [client_1.url],
+                },
+            }
+        ]
+        url = (
+            reverse("v1:safe-apps:list")
+            + f'{"?client_url=pump.com&client_url=safe.com"}'
+        )
+
+        response = self.client.get(path=url, data=None, format="json")
+
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.json(), json_response)
+
+    def test_apps_returned_on_filtered_client_url(self) -> None:
+        client = ClientFactory.create(url="safe.com")
+        client_2 = ClientFactory.create(url="pump.com")
+        safe_app_1 = SafeAppFactory.create(exclusive_clients=(client,))
```

Comment:
Should we add a test entry that has both clients set and should then also be returned?

Reply:
I adjusted the existing test to include such app as well